JSDemos/Demos/DataGrid/CustomNewRecordPosition/jQuery/index.js

```diff
@@ -0,0 +1,74 @@
+$(() => {
+  const newRowPositionOptions = ['first', 'last', 'pageBottom', 'pageTop', 'viewportBottom', 'viewportTop'];
+  const scrollingModeOptions = ['standard', 'virtual'];
+
+  const dataGrid = $('#gridContainer').dxDataGrid({
+    dataSource,
+    showBorders: true,
+    columnAutoWidth: true,
+    editing: {
+      allowAdding: true,
+      allowDeleting: true,
+      allowUpdating: true,
+      confirmDelete: false,
+      useIcons: true,
+      newRowPosition: 'viewportTop',
+    },
+    columns: [
+      {
+        dataField: 'OrderID',
+        allowEditing: false,
+      }, {
+        dataField: 'OrderDate',
+        dataType: 'date',
+        validationRules: [{ type: 'required' }],
+      }, 'ShipName', 'ShipCity', 'ShipPostalCode', 'ShipCountry', {
+        type: 'buttons',
+        buttons: [{
+          icon: 'add',
+          onClick(e) {
+            const changes = dataGrid.option('editing.changes');
+            const key = new DevExpress.data.Guid().toString();
+            changes.push({
+              key,
+              type: 'insert',
+              insertAfterKey: e.row.key,
+            });
+            dataGrid.option('editing.changes', changes);
+            dataGrid.option('editing', {
+              editColumnName: 'OrderDate',
+              editRowKey: key,
+            });
```

Comment:
It is possible to change all options together:
```
dataGrid.option('editing', {
  changes: changes,
  editColumnName: 'OrderDate',
  editRowKey: key,
});
```

Reply:
It doesn't work if I change them together